### DISCUSSION TITLE
Generating a vapid key

### DISCUSSION THREAD

**Discussion by berksmbl** [mastodon/mastodon#24304] [Q&A]
Hello, I'm preparing a one-click installation schema for EasyPanel, and in order for people to easily use this schema, ENV variables need to be prepared beforehand. Is it a correct method to generate VAPID keys in this way?

```js
const crypto = require("crypto");
const curve = crypto.createECDH("prime256v1");
curve.generateKeys();
const vapidPublicKey = curve.getPublicKey("base64");
const vapidPrivateKey = curve.getPrivateKey("base64");
```

**Comment by DejavuMoe** [comment]
No, your method is not necessary. Prepare your [docker-compose.yml](https://github.com/mastodon/mastodon/blob/main/docker-compose.yml) file, then:

```bash
# run twice 
# once as SECRET_KEY_BASE=<browser session secret> 
# once as OTP_SECRET=<multi-factor authentication secret>
 
sudo docker compose run --rm web bundle exec rake secret 

# run once 
# with VAPID_PRIVATE_ KEY=<push notification private key> 
# and VAPID_PUBLIC_KEY=<push notification public key> 

sudo docker compose run --rm web bundle exec rake mastodon:webpush:generate_vapid_key
```

**Reply by berksmbl** [reply]
> No, your method is not necessary. Prepare your [docker-compose.yml](https://github.com/mastodon/mastodon/blob/main/docker-compose.yml) file, then:

I already know that. It's not possible for me to use this method. I am creating a one-click installation scheme for EasyPanel, and in this scheme, ENV variables need to be automatically prepared and defined for people to use. So, I can't execute a command and use the returned data.

**Comment by grrywlsn** [comment]
Did you get a solution for this, @berksmbl ?

**Reply by berksmbl** [reply]
I am currently using this method, but I am not sure if it is the right one.
